### DISCUSSION TITLE
Explicit Valence Error

### DISCUSSION THREAD

**Discussion by heqing-chriscao** [rdkit/rdkit#7090] [Q&A] [answered]
Hello,

When I want to convert an SDF file into a PDBQT file using the following script,
```
molecule_file = '1exd_apatamer_Model_1H.sdf'
file_from_sdf = Chem.SDMolSupplier(molecule_file, removeHs=False)[0]

preparator = MoleculePreparation()
mol_setups = preparator.prepare(file_from_sdf)

for setup in mol_setups:
    pdbqt_string = PDBQTWriterLegacy.write_string(setup)[0]

# Write pdbqt string to file
with open('1exd_apatamer_Model_1H.pdbqt','w') as fw:
    for line in pdbqt_string:
        fw.write(line)
```
an error always occurs:
```
[21:43:07] Explicit valence for atom # 549 H, 2, is greater than permitted
[21:43:07] ERROR: Could not sanitize molecule ending on line 4891
[21:43:07] ERROR: Explicit valence for atom # 549 H, 2, is greater than permitted
```

To my understanding, this implies that the hydrogen atom with the ID 549 forms more than one bond with other atoms. However, when I open the SDF file and examine the bonds section, I observe only a single bond involving hydrogen atom 549. The SDF file, in TXT format, is attached here: [1exd_apatamer_Model_1H.txt](https://github.com/rdkit/rdkit/files/14032017/1exd_apatamer_Model_1H.txt)

I am confused about this discrepancy. Can anyone explain what might be happening and offer assistance? Thank you so much in advance.

**Comment by greglandrum** [comment, marked as answer]
Hi @heqing-chriscao, the RDKit indexes atoms starting from zero, but the SDF indexes starting from 1. So if the RDKit says that there's a problem with H 549, then you need to look for atom 550 in the SDF. That has, in fact, two bonds:

**Reply by heqing-chriscao** [reply]
Thank you very much for your explanation. It is very helpful.
